app/components/graph-editor/events.ts

```diff
@@ -0,0 +1,60 @@
+// File: events.ts
+// Created by: CJ Dimaano
+// Date created: February 7, 2017
+
+
+export type Listener<A extends EventArgs> = (evt?: A) => void;
+
+export type PropertyChangedEventListener<T> = Listener<PropertyChangedEventArgs<T>>;
+
+export class EventArgs {
```

Comment:
Interfaces usually have less coupling and are easier to extend (i.e. don't have to correspond to inheritance hierarchy).